packages/clay-time-picker/src/index.tsx

```diff
@@ -77,6 +77,11 @@ interface IProps
 	 */
 	defaultValue?: Input;
 
+	/**
+	 * Id to be applied to `clay-time-hours` input.
+	 */
+	hoursInputId?: string;
```

Comment:
I think you can just call `id` and bind the id to the field group instead of just the hour input would be better for the screen reader, we need to add the correct ARIA semantics to the group to create the association correctly and that improves accessibility for that component as well.

Adding `hoursInput*` is that it's a not very intuitive name and we don't want the developer to have to know that the id is going to the hour input just that it's an internal implementation level that the developer doesn't have to worry about it.

Reply:
I changed `hoursInputId` to `id`. Clicking on a form label will automatically focus the Time input. The problem with the `for` attribute is that the browser only moves focus on click if it's linked to a proper form element. We have to manually implement a focus change if it maps to a `div`. It's hard to do if the `label` isn't included in the component.

Also it looks like `aria-label` can be customized, we can leave the screen reader announcements to the user.

Reply:
Oh yes you are right, we would need to add a handler to correctly move the focus so it works properly. Maybe we need to start working on #4741 to redesign the default form related components to provide things like this OOTB.